x-pack/plugins/transform/server/lib/alerting/transform_health_rule_type/transform_health_service.ts

```diff
@@ -146,6 +212,29 @@ export function transformHealthServiceProvider(
         }
       }
 
+      if (testsConfig.errorMessages.enabled) {
+        const response = await this.getErrorMessagesReport(transformIds);
+        if (response.length > 0) {
```

Comment:
Once an transform has errors in its messages, how does the alert ever recover? Currently the rule will stay `active` forever won't it, unless you manually deleted an error message from the notifications index.

Reply:
It recovers when all assigned transforms are not in the `failed` state anymore, added in https://github.com/elastic/kibana/pull/128731/commits/65e4c36b3e340a534347dd9817a866f6ca579543